### DISCUSSION TITLE
How do I use Serenata (PHP lsp)?

### DISCUSSION THREAD

**Discussion by mdtrooper** [autozimu/LanguageClient-neovim#1229] [Q&A]
Hi.

I am trying to use https://serenata.gitlab.io/ but it needs before to launch the server (it only has tcp).

Could I add two commands for a language in the config? such as:

```
let g:LanguageClient_serverCommands = {
    \ 'php': [['./distribution.phar --uri=tcp://127.0.0.1:11111'], ['tcp://127.0.0.1:11111']],
```

Regards.

**Comment by martskins** [comment]
That is not possible right now, no. But nothing stops you from writing your own bash script to do that.

Having said that, I'm not sure if there is a more elegant way to achieve this, I can't say I've ever used a tcp only server and I wasn't involved with this at the time it was implemented, so I'm inclined to say your best bet is to just write a bash script, but I'm not 100% sure to be completely honest with you.
